cloudinit/cmd/main.py

```diff
@@ -461,133 +462,150 @@ def main_init(name, args):
     # Any log usage prior to setup_logging above did not have local user log
     # config applied.  We send the welcome message now, as stderr/out have
     # been redirected and log now configured.
-    welcome(name, msg=w_msg)
-    LOG.info("PID [%s] started cloud-init '%s'.", os.getppid(), bootstage_name)
+    welcome(stage.welcome)
+    LOG.info("PID [%s] started cloud-init '%s'.", os.getppid(), stage.welcome)
 
     # re-play early log messages before logging was setup
     for lvl, msg in early_logs:
         LOG.log(lvl, msg)
 
     # Stage 3
     try:
-        init.initialize()
+        init.initialize_filesystem()
     except Exception:
         util.logexc(LOG, "Failed to initialize, likely bad things to come!")
     # Stage 4
-    path_helper = init.paths
-    purge_cache_on_python_version_change(init)
-    mode = sources.DSMODE_LOCAL if args.local else sources.DSMODE_NETWORK
+    init.start_boot()
+    init.track_python_version_change()
 
-    if mode == sources.DSMODE_NETWORK:
+    if stage == stages.network:
         if not os.path.exists(init.paths.get_runpath(".skip-network")):
             LOG.debug("Will wait for network connectivity before continuing")
             init.distro.wait_for_network()
-        existing = "trust"
         sys.stderr.write("%s\n" % (netinfo.debug_info()))
     else:
-        existing = "check"
-        mcfg = util.get_cfg_option_bool(init.cfg, "manual_cache_clean", False)
-        if mcfg:
-            LOG.debug("manual cache clean set from config")
-            existing = "trust"
-        else:
-            mfile = path_helper.get_ipath_cur("manual_clean_marker")
-            if os.path.exists(mfile):
-                LOG.debug("manual cache clean found from marker: %s", mfile)
-                existing = "trust"
-
-        init.purge_cache()
+        init.reset_instance()
 
     # Stage 5
     bring_up_interfaces = _should_bring_up_interfaces(init, args)
     try:
-        init.fetch(existing=existing)
-        # if in network mode, and the datasource is local
+        init.fetch()
```

Comment:
cache_mode is either set at `Init()` instantiation, or discovered at runtime in `Init().cache_mode`